libs/bloom-filter/include/bloom_filter/metadata_store.hpp

```diff
@@ -0,0 +1,135 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2020 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "meta/type_traits.hpp"
+
+#include <cstdint>
+#include <fstream>
+
+namespace fetch {
+namespace bloom {
+namespace details {
+
+struct MetaDataHeader
+{
+  uint64_t version;
+  uint64_t length;
+};
+
+template <typename T>
+meta::IfIsPod<T> WriteToStream(std::ostream &stream, T const &element)
+{
+  auto const *raw = reinterpret_cast<char const *>(&element);
+  stream.write(raw, sizeof(T));
```

Comment:
Shouldn't this should check the stream's `badbit`?

Reply:
Good spot - This whole of this code is temporary at the moment, actually. In case of failure it would be detected by other means. And since we rebuild the whole of the bloom filter at the moment I am inclined not to fix this here at this time.

The next stage here will be for @n-hutton to integrate his changes on and remove this metdata file.